### ISSUE TITLE
8.0.0 how to take a screenshot without blinking viewport

### ISSUE THREAD

**Issue by mahnunchik** [puppeteer/puppeteer#7043]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 8.0.0
* Platform / OS version: macOS
* URLs (if applicable): -
* Node.js version: 14

**What steps will reproduce the problem?**

_Please include code that reproduces the issue._

1. Open browser with changed `defaultViewport`
2. Open page
3. Take a screenshot

```js
const browser = await puppeteer.launch({
  headless: true,
  defaultViewport: {
      // MacBook PRO 13"
      width: 1366,
      height: 900,
      deviceScaleFactor: 2,
    },
});

const page = await browser.newPage();

await page.goto(link);

await page.screenshot({
    type: 'jpeg',
    quality: 80,
  });
```

**What is the expected result?**

Normal screenshot as it was in version 6.0.0

**What happens instead?**

Sometimes it is ok, sometimes page is resized to some small dimension, sometimes screenshot it seems to taken in the middle of page resize O_o

**Comment by mahnunchik** [comment]
Related issues:

* https://github.com/puppeteer/puppeteer/issues/6985
* https://github.com/puppeteer/puppeteer/issues/6921
* https://github.com/puppeteer/puppeteer/issues/6823
* https://github.com/puppeteer/puppeteer/pull/6805
* https://github.com/puppeteer/puppeteer/issues/6991

**Comment by mahnunchik** [comment]
Setting `captureBeyondViewport` to `false` solves the issue: https://github.com/puppeteer/puppeteer/blob/v8.0.0/src/common/Page.ts#L1727-L1732

How to set this option from `page.screenshot`? 🤔

**Comment by mahnunchik** [comment]
Monkey patch:

```sh

sed -i -e "s|captureBeyondViewport: true|captureBeyondViewport: false|g" ./node_modules/puppeteer/lib/cjs/puppeteer/common/Page.js
sed -i -e "s|captureBeyondViewport: true|captureBeyondViewport: false|g"  ./node_modules/puppeteer/lib/esm/puppeteer/common/Page.js
```

**Comment by galenso** [comment]
> Monkey patch:
> 
> ```shell
> sed -i -e "s|captureBeyondViewport: true|captureBeyondViewport: false|g" ./node_modules/puppeteer/lib/cjs/puppeteer/common/Page.js
> sed -i -e "s|captureBeyondViewport: true|captureBeyondViewport: false|g"  ./node_modules/puppeteer/lib/esm/puppeteer/common/Page.js
> ```

this patch is working, but is there a way to fix it from the api? page.screenshot({ captureBeyondViewport: false }) ?

**Comment by mahnunchik** [comment]
@galenso I've mentioned it in the original PR of that feature: https://github.com/puppeteer/puppeteer/pull/6805#issuecomment-812082510

**Comment by lehni** [comment]
The option has been released in `v9.0.0` and it works. I believe this issue can be closed.
